### DISCUSSION TITLE
Plausible is not registering site visits

### DISCUSSION THREAD

**Discussion by fyrak1s** [plausible/analytics#1931] [Self-Hosted Support] [answered]
Hi,

I have a self hosted instance of Plausible, having followed the guides in https://plausible.io/docs/self-hosting.

I have the plausible dashboard accessable over a reverse proxy, I can log in and add my website.

I have also added the line

```
<script defer data-api="/api/event" data-domain="mydomain.org" src="/js/script.js"></script>
```
to the site I want to be analysed and according to the browser inspector it loads the script.

But the real issue is that Plausible registers 0 visitors! It shows just the blinking green dot all the time. I think I have followed the guides thoroughly, but nothing helps to solve the issue.

Thanks in advance for help.

**Comment by ghost** [comment, marked as answer]
You may have to put the full url to where you have plausible installed.

<script defer data-api="https://myplaysuble.com/api/event" data-domain="mydomain.org" src="https://myplausible.com/js/script.js"></script>


I understand that you do not use nginx as a proxy to serve the script and make the call to the event.

**Reply by fyrak1s** [reply]
Wow thanks!... setting `data-api="https://plausible.mydomain.org/api/event"` did the trick in this case!

I did not change `src="/js/script.js"` though, as it seems to work right?

My reverse proxy is caddy. My config for the selfhosted plausible is very simply:
```
plausible.mydomain.org {
    reverse_proxy localhost:8000
}
```

And for the root domain (the site I want to be analysed):
```
mydomain.org {
    [...]

    @plausible path /js/script.js /api/event
    handle @plausible {
        # Change this if you use a different variant of the script
        # e.g. rewrite /js/script.js /js/plausible.outbound-links.js
        rewrite /js/script.js /js/plausible.js
        reverse_proxy https://plausible.io {
            header_up Host {http.reverse_proxy.upstream.hostport}
        }
    }
}
```

**Comment by ghost** [comment]
script.js should also carry the full url.

Better 127.0.0.1 than localhost.

**Reply by fyrak1s** [reply]
I think it works because of the rewrite rules I have defined in caddy?

See my updated comment above.

So basically it takes the script from `https://plausible.io` I think.

**Comment by ghost** [comment]
I don't know how caddy works, so if it already works, that's it.

**Reply by fyrak1s** [reply]
If `/js/script.js` is being accessed on root domain then it redirects that to `https://plausible.io/js/plausible.js`. That is my understanding.

Anyways thanks for helping out! :)

**Comment by ghost** [comment]
Maybe you are using the javascript of the commercial version (plausible.io), but I really don't know how caddy works.

**Reply by fyrak1s** [reply]
I am not sure either, I have simply copied it from their guide https://plausible.io/docs/proxy/guides/caddy :)

Which is also where I've had
```
<script defer data-api="/api/event" data-domain="website.com" src="/js/script.js"></script>
```
from.
